### PR TITLE
Razvan - Homework

### PR DESCRIPTION
**Pull request by razvanm3**

### CODE REVIEW

File: h2-service/src/main/java/dao/OrderDao.java [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
-        String sql = "SELECT o FROM Order o";
-        TypedQuery<Order> query = em.createQuery(sql, Order.class);
-        return query.getResultList();
-    }
-
-}
+package dao;
+
+import com.procourier.model.Order;
+import com.procourier.model.Product;
```
Comment (cribratu) [inline h2-service/src/main/java/dao/OrderDao.java]: Please remove unused imports

File: orders-service-client/src/main/java/http/GetOrderByIdClient.java [hunk 2]
```diff
@@ -1,44 +1,79 @@
...
+            e.printStackTrace();
+        }
+
+        return null;
+    }
+
+
+}
+
+public class GetOrdersClient {
```
Comment (cribratu) [inline orders-service-client/src/main/java/http/GetOrderByIdClient.java]: No need to create different class. You can just rename the previous one and add this new method

File: orders-service-client/src/main/java/http/GetOrderByIdClient.java [hunk 3]
```diff
@@ -1,44 +1,79 @@
...
+    private Gson gson = new Gson();
```
Comment (cribratu) [inline orders-service-client/src/main/java/http/GetOrderByIdClient.java]: You can make son and httpClient final as well

File: orders-service-client/src/main/java/http/GetOrderByIdClient.java [hunk 4]
```diff
@@ -1,44 +1,79 @@
...
+    private HttpClient httpClient;
+
+    public GetOrdersClient(){
+        httpClient = HttpClient.newBuilder()
+                .version(HttpClient.Version.HTTP_1_1)
+                .build();
+    }
+
+    public List<Order> getOrders(){
+        Type OrderList = new TypeToken<ArrayList<Order>>(){
```
Comment (cribratu) [inline orders-service-client/src/main/java/http/GetOrderByIdClient.java]: final

File: orders-service-client/src/main/java/main/Main.java [hunk 5]
```diff
@@ -1,13 +1,18 @@
...
+package clientapp;
+
+import com.procourier.model.Order;
+import http.GetOrderByIdClient;
+
+import java.util.List;
+
+public class Main {
+    public static void main(String[] args) {
+        final GetOrderByIdClient client = new GetOrderByIdClient();
```
Comment (cribratu) [inline orders-service-client/src/main/java/main/Main.java]: Why instantiating GetOrderByIdClient twice?

File: orders-service/src/main/java/com/procourier/handlers/GetOrdersHandler.java [hunk 6]
```diff
@@ -1,4 +1,33 @@
-package com.procourier.handlers;
-
-public class GetOrdersHandler {
-}
+package com.procourier.handlers;
+
+import com.google.gson.Gson;
+import com.procourier.model.Order;
+import com.procourier.model.repository.OrderRepository;
```
Comment (cribratu) [inline orders-service/src/main/java/com/procourier/handlers/GetOrdersHandler.java]: Please remove unused imports

File: orders-service/src/main/java/com/procourier/handlers/GetOrdersHandler.java [hunk 7]
```diff
@@ -1,4 +1,33 @@
...
+import service.OrderService;
+import spark.Request;
+import spark.Response;
+import spark.Route;
+import java.util.List;
+
+import static java.lang.Long.valueOf;
+
+public class GetOrdersHandler implements Route{
+    private OrderService orderService;
```
Comment (cribratu) [inline orders-service/src/main/java/com/procourier/handlers/GetOrdersHandler.java]: final for every field/local variable whose value never changes. Keep this in mind although in our examples we didn't do this all the time. It's a good practice

File: orders-service/src/main/java/com/procourier/handlers/GetOrdersHandler.java [hunk 8]
```diff
@@ -1,4 +1,33 @@
...
+    private Gson gson = new Gson();
+
+    public GetOrdersHandler(OrderService orderService) {
+        this.orderService = orderService;
+    }
+
```
Comment (cribratu) [inline orders-service/src/main/java/com/procourier/handlers/GetOrdersHandler.java]: [Cosmetic] Too many empty lines

File: orders-service/src/main/java/com/procourier/app/OrderServiceMain.java [hunk 9]
```diff
@@ -1,52 +1,55 @@
...
+
+        final Address cosminAddress = new Address("DJ", "Craiova", "Ileana Cosanzeana");
+        final Buyer buyer = new Buyer("Cosmin", cosminAddress);
+
+        final Address courierAddress = new Address("OT", "Slatina", "Cantemir");
+        final Courier courier = new Courier("ExpressFan", courierAddress,
+                5L, 7L);
+
+
+        final List<OrderLine> products = List.of(new OrderLine(new Product(20L, "Jeans", "Men Jeans", 22L), 4L));
```
Comment (cribratu) [inline orders-service/src/main/java/com/procourier/app/OrderServiceMain.java]: static import here and in every other class where appropriate

**Comment by cribratu** [comment]
No need to implement the changes now, it was just an example of how things work on a real project.
👍

**Comment by razvanm3** [comment]
Thank you for the feedback!

**Pull request closed without merging**